env0/resource_provider.go

```diff
@@ -0,0 +1,146 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceProvider() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceProviderCreate,
+		ReadContext:   resourceProviderRead,
+		UpdateContext: resourceProviderUpdate,
+		DeleteContext: resourceProviderDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceProviderImport},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:             schema.TypeString,
+				Description:      `type of the provider registry (Match pattern: ^[0-9a-z](?:[0-9a-z-]{0,30}[0-9a-z])?$). Your provider’s type is essentially it’s name, and should match your provider’s files. For example, if your binaries look like terraform-provider-aws_1.1.1_linux_amd64.zip, than your provider’s type should be aws.`,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: NewRegexValidator(`^[0-9a-z](?:[0-9a-z-]{0,30}[0-9a-z])?$`),
```

Comment:
I hope this regex is correct. I decided on this regex after "playing" with the UI.

Reply:
this is the regex we use in the UI 
`const PROVIDER_TYPE_INVALID_CHAR_REGEX = /^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$/;`

Reply:
we also have a max characters validation, we limit it to 32
`const MAX_CHARS = 32`